modules/scylladb/scylladb.go

```diff
@@ -0,0 +1,144 @@
+	return func(req *testcontainers.GenericContainerRequest) error {
+		cf := testcontainers.ContainerFile{
+			HostFilePath:      configFile,
+			ContainerFilePath: "/etc/scylla/scylla.yaml",
+			FileMode:          0o755,
+		}
+		req.Files = append(req.Files, cf)
+
+		return nil
+	}
+}
+
+// WithShardAwareness enable shard-awareness in the ScyllaDB container so you can use the `19042` port.
+func WithShardAwareness() testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		req.ExposedPorts = append(req.ExposedPorts, shardAwarePort)
+		req.WaitingFor = wait.ForAll(req.WaitingFor, wait.ForListeningPort(shardAwarePort))
+		return nil
+	}
+}
+
+// WithAlternator enables the Alternator (DynamoDB Compatible API) service in the ScyllaDB container.
+// It will set the "alternator-port" parameter to the specified port.
+// It will also set the "alternator-write-isolation" parameter to "always" as a command line argument to the container.
+func WithAlternator(alternatorPort int) testcontainers.CustomizeRequestOption {
+	strPort := strconv.Itoa(alternatorPort)
+
+	return func(req *testcontainers.GenericContainerRequest) error {
+		setCommandFlag(req, "--alternator-port", strPort)
+		setCommandFlag(req, "--alternator-write-isolation", "always")
+		req.ExposedPorts = append(req.ExposedPorts, strPort)
+		req.WaitingFor = wait.ForAll(req.WaitingFor, wait.ForListeningPort(nat.Port(strPort)))
+		return nil
+	}
+}
+
+// WithCustomCommand sets a custom command with a value for the ScyllaDB container.
+// This is an option to replace the default command with a custom one.
+// See more [here](https://opensource.docs.scylladb.com/stable/operating-scylla/procedures/tips/best-practices-scylla-on-docker.html)
+func WithCustomCommand(command, value string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		setCommandFlag(req, command, value)
+		return nil
+	}
+}
+
+// ConnectionHost returns the host and port of the Scylladb container with the default port
+// and obtaining the host and exposed port from the container
+// Eg: "host:port" -> "localhost:9042" -> "localhost:19042" -> "localhost:8000"
+func (c Container) ConnectionHost(ctx context.Context, port int) (string, error) {
+	host, err := c.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	containerPort, err := c.MappedPort(ctx, nat.Port(strconv.Itoa(port)))
+	if err != nil {
+		return "", err
+	}
+
+	return host + ":" + containerPort.Port(), nil
+}
+
+// Run starts a ScyllaDB container with the specified image and options
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*Container, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        img,
+		ExposedPorts: []string{port},
+		Cmd: []string{
+			"--developer-mode=1",
+			"--overprovisioned=1",
+		},
+		WaitingFor: wait.ForAll(
+			wait.ForLog(".*initialization completed.").AsRegexp(),
```

Comment:
question: log checks can be fragile, is this actually needed or does the listen / exec check cover all the bases?